src/components/goals/GoalCard.tsx

```diff
@@ -0,0 +1,215 @@
+        <DialogTrigger asChild>
+          <Card className="h-48 w-64 cursor-pointer bg-gradient-to-br from-blue-50 to-indigo-50 transition hover:scale-105 hover:shadow-md">
+            <CardContent className="flex h-full flex-col justify-between p-4">
+              <div className="flex items-center justify-between">
+                <div className="rounded-full bg-indigo-50 p-3">
+                  <IconComponent size={24} className="text-indigo-600" />
+                </div>
+                <p className="text-sm text-gray-500">
+                  {new Date(goal.targetDate).toLocaleDateString()}
+                </p>
+              </div>
+              <div className="mt-2">
+                <p className="line-clamp-2 text-lg font-semibold text-gray-700">
+                  {goal.title}
+                </p>
+                <Progress value={progressPercentage} className="mt-2" />
+                <div className="mt-1 text-sm text-gray-600">
+                  ${goal.currentAmount.toLocaleString()} / $
+                  {goal.targetAmount.toLocaleString()}
+                </div>
+              </div>
+            </CardContent>
+          </Card>
+        </DialogTrigger>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>Edit {goal.title} Goal</DialogTitle>
+          </DialogHeader>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <Input
+              {...register("title", { required: "Title is required" })}
+              placeholder="Goal Title"
+              defaultValue={goal.title}
+              className="mb-4"
+            />
+            {errors.title && (
+              <p className="mb-4 mt-2 text-red-500">{errors.title.message}</p>
+            )}
+            <Input
+              {...register("targetAmount", {
+                valueAsNumber: true,
+                required: "Target amount is required",
+                min: {
+                  value: 0.01,
+                  message: "Target amount must be greater than 0",
+                },
+              })}
+              type="number"
+              step={0.01}
+              placeholder={`Current Target: $${goal.targetAmount}`}
+              className="mb-4"
+            />
+            {errors.targetAmount && (
+              <p className="mb-4 mt-2 text-red-500">
+                {errors.targetAmount.message}
+              </p>
+            )}
+            <Input
+              {...register("currentAmount", {
+                valueAsNumber: true,
+                required: "Current amount is required",
+                min: {
+                  value: 0,
+                  message: "Current amount must be 0 or greater",
+                },
+              })}
+              type="number"
+              step={0.01}
+              placeholder={`Current Progress: $${goal.currentAmount}`}
+              className="mb-4"
+            />
+            {errors.currentAmount && (
```

Comment:
It should be possible to make these errors iterative, so given any kind of error, it's placed under the input. that should help clean up some real estate.